d3a_interface/area_validator.py

```diff
@@ -18,17 +18,22 @@
 
 from d3a_interface.constants_limits import ConstSettings
 from d3a_interface.device_validator import validate_range_limit
+from d3a_interface.exceptions import D3ADeviceException
 
 
 GeneralSettings = ConstSettings.GeneralSettings
 AreaSettings = ConstSettings.AreaSettings
 
 
 def validate_area(**kwargs):
-    if ("grid_fee_percentage" in kwargs and kwargs["grid_fee_percentage"] is not None):
+    if "grid_fee_percentage" in kwargs and kwargs["grid_fee_percentage"] is not None:
         error_message = {"misconfiguration": [f"grid_fee_percentage should be in between "
                                               f"{AreaSettings.GRID_FEE_PERCENTAGE_LIMIT.min} & "
                                               f"{AreaSettings.GRID_FEE_PERCENTAGE_LIMIT.max}."]}
         validate_range_limit(AreaSettings.GRID_FEE_PERCENTAGE_LIMIT.min,
                              kwargs["grid_fee_percentage"],
                              AreaSettings.GRID_FEE_PERCENTAGE_LIMIT.max, error_message)
+
+    if "grid_fee_constant" in kwargs and kwargs["grid_fee_constant"] is not None:
+        if kwargs["grid_fee_constant"] < 0:
+            raise D3ADeviceException("Grid fee constant should be a positive value.")
```

Comment:
Shouldn't we also restrict user by setting the ridiculously high value  like we have done for device parameters ?

Reply:
Agreed, setting the upper limit to 200 cts/kWh. Thanks